### PR TITLE
Add description of integration of Magnus

### PR DESCRIPTION
**Pull request by code28**

### CODE REVIEW

File: doc/eID/eid-integration-magnus.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# UseID Integration
+
+To integrate UseID and mobile app and web widget, the following steps are necessary for services already having an existing eID integration.
```
Comment (mpanne) [inline doc/eID/eid-integration-magnus.md:3]: Typo. "To integrate UseID mobile app and web widget".

File: doc/eID/eid-integration-magnus.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+and including this script in the page:
+
+```html
+<script async src="https://useid.ds4g.net/widget.js"></script>
+```
+
+As value for `data-tc-token-url` the already known tcTokenURL is used.
+
+> Previously, here was a link including the tcTokenURL as query parameter for the user to click to open the app. Now a widget is displayed, which can be scanned by the user with the smartphone to start the eID scanning in the UseID mobile app.
```
Comment (mpanne) [inline doc/eID/eid-integration-magnus.md:23]: I would rephrase it to something like: 
"The eService usually presents a link to the user which points to the eID client application (e.g. AusweisApp2) including the tcTokenURL as a query parameter. With UseID, this is replaced by the web widget displaying a QR code which can be scanned to start the UseID mobile app."

File: doc/eID/eid-integration-magnus.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+### 2. ID process
+
+The process of scanning the eID has not changed for the eService. Changes for the users are in using the UseID mobile app and their smartphone as scanning devices.
+
+### 3. Success and Refresh
+
+The user is returned to the `refreshAddress` , which is provided in the TC token, on the device displaying the widget.
```
Comment (mpanne) [inline doc/eID/eid-integration-magnus.md:31]: I would maybe add "The user is returned to the refreshAddress of the eService, ...

File: doc/eID/eid-integration-magnus.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+## Security
+
+Instead of staying on the same device, a QR code is displayed to be scanned by the user's smartphone. At success, the user returns to page displaying the widget and gets forwarded to the refresh address. The embedded widget (locally in the browser) and the smartphone establish an end-to-end encryption channel so that any communication going through UseID servers can only interpreted by the user's devices; especially `tcTokenURL` and `refreshAddress` are never exposed to UseID servers.
```
Comment (mpanne) [inline doc/eID/eid-integration-magnus.md:35]: Typo:
"Instead of staying on the same device, a QR code is displayed to be scanned by the user's smartphone. At success, the user returns to **the** page displaying the widget and gets forwarded to the refresh address. The embedded widget (locally in the browser) and the smartphone establish an end-to-end encryption channel so that any communication going through UseID servers can only **be** interpreted by the user's devices; especially `tcTokenURL` and `refreshAddress` are never exposed to UseID servers."

**Review by mpanne — APPROVED** [review]
Cool integration guide, thanks!

File: doc/eID/eid-integration-magnus.md (hunk 2 above)
Comment (code28) [reply doc/eID/eid-integration-magnus.md:23]: Sound way better! 👍

File: doc/eID/eid-integration-magnus.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+The process of scanning the eID has not changed for the eService. Changes for the users are in using the UseID mobile app and their smartphone as scanning devices.
+
+### 3. Success and Refresh
+
+The user is returned to the `refreshAddress` of the eService, which is provided in the TC token, on the device displaying the widget.
+
+## Security
+
+Instead of staying on the same device, a QR code is displayed to be scanned by the user's smartphone. At success, the user returns to the page displaying the widget and gets forwarded to the refresh address. The embedded widget (locally in the browser) and the smartphone establish an end-to-end encrypted channel so that any communication going through UseID servers can only be interpreted by the user's devices; especially `tcTokenURL` and `refreshAddress` are never exposed to UseID servers.
```
Comment (pgurusinga) [inline doc/eID/eid-integration-magnus.md:35]: I am wondering, the QR Code would be still valid after successful identification via mobile app? should we also invalidate the QR code as well?

Comment (code28) [reply doc/eID/eid-integration-magnus.md:35]: Yes, I think so, too. We need to keep an eye on the handling of sessions, when to start and delete them, and then it should be fine.

Comment (pgurusinga) [reply doc/eID/eid-integration-magnus.md:35]: 1. why does refresh address more crucial?
2. especially tcTokenURL and refreshAddress are never exposed to UseID servers. I am a bit confused by this sentence.

let's say, this is an `tcTokenUrl`
https://www.foo.bar/eid-server-demo-app/result/getTcToken.html;jsessionid=RANDOMSESSIONID and it will return 

```xml
<TCTokenType>
	  <ServerAddress>https://bar.foo.de:443</ServerAddress>
	  <SessionIdentifier>SESSIONIDENTIFIER</SessionIdentifier>
	  <RefreshAddress>https://www.foo.bar:443/eid-server-demo-app/result/response.html;jsessionid=RANDOMSESSIONID</RefreshAddress>
	  <Binding>urn:liberty:paos:2006-08</Binding>
	  <PathSecurity-Protocol>urn:ietf:rfc:4279</PathSecurity-Protocol>
	  <PathSecurity-Parameters>
	  	<PSK>FOOBAR</PSK>
	  </PathSecurity-Parameters>
</TCTokenType>
```

so it already includes RefreshAddress, why do we need both? I am a bit confused...

Comment (code28) [reply doc/eID/eid-integration-magnus.md:35]: What do you mean by "why do we need both"? Maybe we can have a quick chat on that, that's easier than writing?
